### ISSUE TITLE
Error in ^

### ISSUE THREAD

**Issue by dpsanders** [JuliaIntervals/IntervalArithmetic.jl#164]
Reported in #155 by @lbenet:

```
julia> using IntervalArithmetic

julia> @format full
6

julia> a = 0..3
Interval(0.0, 3.0)

julia> a^-2   # WRONG!! Should be --> Interval(0.1111111111111111, Inf)
Interval(0.11111111111111109, Inf)

julia> @which a^-2
^(a::Interval{Float64}, x::Integer) in IntervalArithmetic at /Users/benet/.julia/dev/IntervalArithmetic/src/intervals/functions.jl:10
Using the function that actually executes a^-2 (IntervalArithmetic/src/intervals/functions.jl:10) I get the following:

julia> IntervalArithmetic.atomic(Interval{Float64}, IntervalArithmetic.big53(a)^-2)  # correct answer!!
Interval(0.1111111111111111, Inf)
```

**Comment by lbenet** [comment]
Thanks for opening this issue!

**Comment by eeshan9815** [comment]
Interestingly this doesn't give the inaccurate result. (julia0.7-beta)
```
julia> @format full
6

julia> a = 0..3
Interval(0.0, 3.0)

julia> b = -2
-2

julia> a^b
Interval(0.1111111111111111, Inf)
```

**Comment by lbenet** [comment]
The problem appears in Julia 0.7; did you tried that?

**Comment by eeshan9815** [comment]
Yeah, this is on the latest 0.7 beta

**Comment by eeshan9815** [comment]
```
julia> a = 0..3
Interval(0.0, 3.0)

julia> b = -2
-2

julia> a^-2
Interval(0.11111111111111109, Inf)

julia> a^b
Interval(0.1111111111111111, Inf)
```

**Comment by dpsanders** [comment]
Ouch... This seems to be the specialised small integer power biting again.

**Comment by eeshan9815** [comment]
What seems to be the problem? Some internal optimisation gone wrong for intervals?

**Comment by dpsanders** [comment]
I believe this is due to
https://github.com/JuliaLang/julia/pull/24240

which makes `X^-n` do `inv(X)^n`. 

I'm not sure what to do about this, if anything.

**Comment by dpsanders** [comment]
Apparently we just have to override the `literal_pow` function for `Interval`s.

**Comment by dpsanders** [comment]
The solution seems to be
```
Base.literal_pow(::typeof(^), x::Interval{T}, ::Val{p}) where {T,p} = x^p
```

**Comment by dpsanders** [comment]
Should be fixed by #173.
